src/DaemonExec.ts

```diff
@@ -0,0 +1,244 @@
+	} catch (err) {
+		console.log(err)
+		return []
+	}
+}
+
+const listContainers = async () => {
+	try {
+		const containers = await docker.listContainers()
+		return containers
+	} catch (err) {
+		console.log(err)
+		return []
+	}
+}
+
+const importImage = async (name: string) => {
+	try {
+		const stream = await docker.pull(name)
+		return stream
+	} catch (err) {
+		console.log(err)
+	}
+}
+
+//volumePairs should be of the form [[hostSrc,containerDest]]
+const createContainer = async (
+	image: string,
+	command: string[],
+	inputDir: string,
+	stdinFile: string,
+	outputDir: string,
+	stdoutFile: string,
+	errorDir: string,
+	stderrFile: string,
+	volumePairs: [string, string][],
+) => {
+	try {
+		const volumeJson: any = {}
+		for (const volumePair of volumePairs) {
+			volumeJson[`${volumePair[1]}`] = {}
+		}
+		const volumeArray = volumePairs.map((el) => el[0] + ':' + el[1])
+		const container = await docker.createContainer({
+			Image: image,
+			Cmd: command,
+			Volumes: volumeJson,
+			HostConfig: {
+				Binds: volumeArray,
+			},
+			AttachStdin: true,
+			AttachStdout: true,
+			AttachStderr: true,
+			Tty: false,
+			OpenStdin: true,
+			StdinOnce: true,
+		})
+		const stream = await container.attach({
+			hijack: true,
+			stream: true,
+			stdin: true,
+			stdout: true,
+			stderr: true,
+		})
+		let stdoutStream: any = process.stdout
+		if (outputDir != '' && stdoutFile != '') {
+			await fs.mkdir(outputDir, { recursive: true }, (err) => {
+				if (err) {
+					console.log(err)
+					throw err
+				}
+			})
+			stdoutStream = fs.createWriteStream(`${outputDir}/${stdoutFile}`)
+		}
+		let stderrStream: any = process.stderr
+		if (errorDir != '' && stderrFile != '') {
+			await fs.mkdir(errorDir, { recursive: true }, (err) => {
+				if (err) {
+					console.log(err)
+					throw err
+				}
+			})
+			stderrStream = fs.createWriteStream(`${errorDir}/${stderrFile}`)
+		}
+		container.modem.demuxStream(stream, stdoutStream, stderrStream)
+
+		if (inputDir != '' && stdinFile != '') {
+			const readStream = fs.createReadStream(
+				`${inputDir}/${stdinFile}`,
+				'binary',
+			)
+			readStream.pipe(stream)
+		}
+
+		return container
+	} catch (err) {
+		console.log(err)
+		return null
+	}
+}
+
+const stopContainer = async (container: any) => {
+	try {
+		await container.stop()
+	} catch (err) {
+		console.log(err)
+	}
+}
+
+const removeContainer = async (container: any) => {
+	try {
+		await container.remove()
+	} catch (err) {
+		console.log(err)
+	}
+}
+
+const runContainer = async (container: any) => {
+	container.start(async (err: any) => {
+		if (err) {
+			console.log(err)
+		} else {
+			console.log('started')
+			container.wait(async (err: any, data: any) => {
+				if (err) {
+					console.log(err)
+				}
+				console.log('container end: ', data)
+				await container.remove()
+				console.log('removed')
+			})
+		}
+	})
+}
+
+//This method is only an example of how to run a command. This is not part of the API
```

Comment:
This is a great example, we should avoid leaving code in comments though. Can you move this into a separate file like `src/docker-demo.ts` and add a `#!` line so we can directly run it?